src/platform/plugins/shared/discover/public/application/main/components/top_nav/app_menu_actions/get_share.tsx

```diff
@@ -26,125 +26,151 @@ export const getShareAppMenuItem = ({
   discoverParams: AppMenuDiscoverParams;
   services: DiscoverServices;
   stateContainer: DiscoverStateContainer;
-}): AppMenuActionPrimary => {
-  return {
-    id: AppMenuActionId.share,
-    type: AppMenuActionType.primary,
-    controlProps: {
-      label: i18n.translate('discover.localMenu.shareTitle', {
-        defaultMessage: 'Share',
-      }),
-      description: i18n.translate('discover.localMenu.shareSearchDescription', {
-        defaultMessage: 'Share Discover session',
-      }),
-      iconType: 'share',
-      testId: 'shareTopNavButton',
-      onClick: async ({ anchorElement }) => {
-        const { dataView, isEsqlMode } = discoverParams;
+}): AppMenuActionPrimary[] => {
+  if (!services.share) {
+    return [];
+  }
 
-        if (!services.share) {
-          return;
-        }
+  const shareExecutor = async ({
```

Comment:
Can Share modal have more tabs other than "Link" now?

<img width="549" alt="Screenshot 2025-05-07 at 17 06 00" src="https://github.com/user-attachments/assets/5b8f1bbc-7638-4ae9-ad07-feece51791da" />

Reply:
General yes, for discover I suppose not.